src/Generator/IdGenerator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\MediaBundle\Generator;
+
+use Sonata\MediaBundle\Model\MediaInterface;
+
+/**
+ * @final since sonata-project/media-bundle 3.21.0
+ */
+class IdGenerator implements GeneratorInterface
```

Comment:
You can make it final rightaway, it's not a BC-break

Reply:
It's actually extended by the deprecated `DefaultGenerator` right now.
https://github.com/sonata-project/SonataMediaBundle/pull/1688/files#diff-fcd815cd554f9f3cf32406cc1f5b1831R21

Reply:
You can still make this final  by proxing the old methods

Reply:
Or better, use a `class_alias`, like in this PR: https://github.com/doctrine/persistence/pull/71

Reply:
@bigfoot90 don't hesitate to ask me if you want some assistance / an example for this.